src/lib/fountain/fountain_decoder_sink.h

```diff
@@ -33,13 +45,21 @@ class fountain_decoder_sink
 		return _chunkSize;
 	}
 
-	bool store(const FountainMetadata& md, const std::vector<uint8_t>& data)
+	std::string get_filename(const FountainMetadata& md) const
 	{
-		std::string file_path = fmt::format("{}/{}", _dataDir, get_filename(md));
-		OUTSTREAM f(file_path, std::ios::binary);
-		f.write((char*)data.data(), data.size());
-		if (_logWrites)
-			printf("%s\n", file_path.c_str());
+		return fmt::format("{}.{}", md.encode_id(), md.file_size());
+	}
+
+	bool store(const FountainMetadata& md, fountain_decoder_stream& s)
+	{
+		if (_onStore)
+		{
+			auto res = s.recover();
+			if (!res)
+				return false;
+			_onStore(get_filename(md), *res);
+			mark_done(md);
```

Comment:
Notably, caller *must* explicitly call `mark_done()` if the function callback isn't in use here.